docs/rfc/rfc-005-authentication.md

```diff
@@ -0,0 +1,132 @@
+
+### 4. Extensible Credential Mechanisms
+   
+Currently, the Cosmos SDK primarily supports accounts backed by public/private key pairings. However, there is a growing
+need to design and implement an extensible system that can support abstracted credentials, which are not limited to
+public/private key pairings. This would enable the development of more advanced and flexible authentication systems that
+cater to a broader range of use cases and security requirements.
+
+To address these problems, the Cosmos SDK needs to develop an authentication system that is closely integrated with the
+proposed account system, providing a robust and extensible framework for managing authentication credentials and supporting
+a variety of credential mechanisms.
+
+## Implementation
+
+We have initially mentioned that `x/accounts` only provides for a way to deploy accounts, but they're not backed by any
+credential, which means no state transition on behalf of the account can be incepted from a TX.
+
+`x/authn` is meant to provide the link between accounts and their TX credentials.
+Specifically `x/authn` defines the following `tx` gRPC interface:
+
+```protobuf
+// Msg defines the Msg service.
+service Msg {
+  rpc CreateAuthenticatedAccount(MsgCreateAuthenticatedAccount) returns (MsgCreateAuthenticatedAccountResponse) {};
+  rpc UpdateCredentials(MsgUpdateCredentials) returns (MsgUpdateCredentialsResponse) {};
+  rpc DeleteCredentials(MsgDeleteCredentials) returns (MsgDeleteCredentials) {};
+}
+message MsgCreateAuthenticatedAccount {
+  string sender = 1;
+  google.Protobuf.Any credential = 2;
+  accounts.MsgDeploy deploy_msg = 3;
+}
+message MsgCreateAuthenticatedAccountResponse {
+  accounts.MsgDeployResponse deploy_response = 1;
+}
+message MsgUpdateCredentials {
+  string sender = 1;
+  string kind = 2;
+  google.Protobuf.Any new_credential = 3;
+}
+message MsgUpdateCredentialsResponse {}
+message MsgDeleteCredentials {
+  string sender = 1;
+}
+message MsgDeleteCredentialsResponse {}
+```
+
+### MsgCreateAuthenticatedAccount
+
+This message contains an opaque credential defined as `google.Protobuf.Any`, alongside an `accounts.MsgDeploy` request.
+This creates a new account and couples it with a credential.
+
+### MsgUpdateCredentials & MsgDeleteCredentials
+
+The former allows for credentials of an account to be updated the latter destroys credentials for an account, making it
+effectively impossible for the account to send state transitions from a TX forever, unless the account has logic to again
+update its credentials.
+
+### The credential interface
+
+The credential interface, represented in our gRPC `tx` interface as a `google.Protobuf.Any`, is implemented by any type
+which satisfies the following interface:
+
+```go
+package authn
+type Credential[T any, PT interface{ *T; proto.Message }] interface {
+	*T
+	VerifySignedBytes(msgBytes []byte, signature []byte) bool
```

Comment:
`VerifySignedBytes` implies a signature. In this model we may not need a signature at all (for example, when providing a zk proof). We can make this more generic like:

```suggestion
	Verify(msgBytes []byte, verification Any) bool
```